src/arrayMethodSort.js

```diff
@@ -5,7 +5,62 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const inputArray = [...this];
+
+    if (compareFunction === undefined) {
+      for (let l = 0; l < inputArray.length; l++) {
+        inputArray[l] += '';
+      }
+
+      for (let i = 0; i <= this.length - 1; i++) {
+        for (let j = 0; j < this.length - 1; j++) {
+          for (let k = 0; k <= inputArray[j].length - 1; k++) {
+            const firstValue = inputArray[j];
+            const secondValue = inputArray[j + 1];
+
+            if (firstValue.charCodeAt(k) === secondValue.charCodeAt(k)) {
+              continue;
+            } else if (firstValue.charCodeAt(k) > secondValue.charCodeAt(k)) {
+              inputArray[j + 1] = firstValue;
+              inputArray[j] = secondValue;
+              break;
+            } else {
+              break;
+            }
+          }
+        }
+      }
```

Comment:
Looks too complicated and a lot of duplication. You can create a separate function same as in sort method [documentation ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)and set it as a default value for compareFunction

function compareFn(a, b) {
  if (a is less than b by some ordering criterion) {
    return -1;
  } else if (a is greater than b by the ordering criterion) {
    return 1;
  }
  // a must be equal to b
  return 0;
}

Reply:
Separate function for comparison created